src/Middleware/HttpLogging/src/HttpLoggingMiddleware.cs

```diff
@@ -144,63 +184,43 @@ private async Task InvokeInternal(HttpContext context)
         {
             var response = context.Response;
 
-            if (loggingFields.HasFlag(HttpLoggingFields.ResponseStatusCode) || loggingFields.HasFlag(HttpLoggingFields.ResponseHeaders))
+            if (logContext.IsAnyEnabled(HttpLoggingFields.ResponsePropertiesAndHeaders))
             {
                 originalUpgradeFeature = context.Features.Get<IHttpUpgradeFeature>();
 
                 if (originalUpgradeFeature != null && originalUpgradeFeature.IsUpgradableRequest)
                 {
-                    loggableUpgradeFeature = new UpgradeFeatureLoggingDecorator(originalUpgradeFeature, response, options._internalResponseHeaders, loggingFields, _logger);
+                    loggableUpgradeFeature = new UpgradeFeatureLoggingDecorator(originalUpgradeFeature,
+                        logContext, options._internalResponseHeaders, _interceptors, _logger);
 
                     context.Features.Set<IHttpUpgradeFeature>(loggableUpgradeFeature);
                 }
             }
 
-            if (loggingFields.HasFlag(HttpLoggingFields.ResponseBody))
+            if (loggingFields.HasFlag(HttpLoggingFields.ResponseBody) || _interceptors.Length > 0)
```

Comment:
Why is the response body captured if there is an interceptor? Is there a situation where this wouldn't be wanted?

Add comment?

Reply:
The response body isn't captured until after interceptors run, but we need to hook it in advance so that an interceptor can decide to capture the body or not.

Reply:
That seems like a limitation of interceptors. Someone might want to redact fields with an interceptor, but an interceptor forces the body to be captured which can have a big performance overhead.

Have you considered a property on the interceptor interface to control whether the body is captured? `bool CanCaptureResponseBody` or `bool SuppressCaptureResponseBody`

Reply:
We're not always capturing the response body, we're only shimming it so we can see when it starts and make the decision to capture it. This should not cause significant overhead.